### PR TITLE
Refact: don't import store explicitly at the utils/backend.js

### LINKED ISSUE #707: Don't omit exceptions thrown from backend fetch
https://github.com/aeternity/superhero-ui/blob/a60c0d4bad5fcf85a22d47753d5032d12e1d5c3d/src/utils/backend.js#L5

### PR DESCRIPTION
**Pull request by shapkarin**
issue: #909 
ref: https://github.com/aeternity/superhero-ui/pull/993

upd:
closes #707
closes aeternity/superhero-wallet#548

### CODE REVIEW

**Comment by shapkarin** [comment]
@Liubov-crypto please test that:
- open the app
- switch off the internet to produce the request error
- check if the `Maintenance page` is showed

**Comment by shapkarin** [comment]
@mradkov basically all is fine and it's enough to integrate the SSR because the main goal of that task is to not import store explicitly at the `utils/backend.js`.
I already tested [that](https://github.com/aeternity/superhero-ui/pull/1003#issuecomment-768492597) but we can wait for @Liubov-crypto to make sure.

My previous PR https://github.com/aeternity/superhero-ui/pull/993 is marked as draft but later we can continue with more deeply approach that present at that PR or connect SDK and swagger to replace the `utils/backend.js` with SDK methods that was grabbed from the swagger file. Me or @davidyuk can explain that, if you are not aware about the last.

**Comment by shapkarin** [comment]
thanks to @davidyuk 
it also can use `window.addEventListener("unhandledrejection", function (event) {` instead to catch the manually defined `throw new error` from the fetch.

**Comment by shapkarin** [comment]
> thanks to @davidyuk
> it can also use `window.addEventListener("unhandledrejection", function (event) {` instead to catch the manually defined `throw new error` from the fetch.

refs: 
https://github.com/aeternity/superhero-ui/issues/707
https://github.com/aeternity/superhero-wallet/issues/548

**Comment by shapkarin** [comment]
there is some blockers with `unhandledrejection` so I will leave it with `eventBus`

File: src/utils/backend.js [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
   try {
     return promise.then((res) => {
       if (!res) {
-        store.commit('setBackendStatus', false);
+        EventBus.$emit('maintenance');
         return null;
       }
-      store.commit('setBackendStatus', true);
-      if (!res.ok) throw new Error(`Request failed with ${res.status}`);
+      // store.commit('setBackendStatus', true);
```
Comment (mradkov) [inline src/utils/backend.js:11]: When does the maintenance page return back to the main feed, if you comment this out?

Comment (shapkarin) [reply src/utils/backend.js:11]: I will remove that comment. It works with eventBus now:
next line is `EventBus.$emit('maintenance');`
and at the App.vue i have a handler:
```js
EventBus.$on('maintenance', () => this.$router.push({ name: 'maintenance' }));
```

Comment (shapkarin) [reply src/utils/backend.js]: so basically it's:
https://github.com/aeternity/superhero-ui/blob/f55bc1918c0e51011772c999b653dae9bef68ae7/src/utils/backend.js#L8
and
https://github.com/aeternity/superhero-ui/blob/f55bc1918c0e51011772c999b653dae9bef68ae7/src/App.vue#L56

**Comment by Liubov-crypto** [comment]
> @Liubov-crypto please test that:
> 
> * open the app
> * switch off the internet to produce the request error
> * check if the `Maintenance page` is showed

Followed steps described above. Got this in Chrome (the picture of the blue diamond was not loaded):

ууу

Also, when the Internet appears there is no automatic redirection to the main page, the user remains on the maintenance page:
rr

FF - maintenance text and the picture of the blue diamond was not loaded:
мозіла

Brave - maintenance text and the picture of the blue diamond was not loaded:
брейв1

**Comment by shapkarin** [comment]
@Liubov-crypto thanks, need to investigate what's going on in Brave

**Comment by shapkarin** [comment]
currently I got that page on error, but from time to time it not appears. need to investigate in any case...
Снимок экрана 2021-01-28 в 11 56 19

**Comment by shapkarin** [comment]
@Liubov-crypto try to reproduce it that way in Brave:
- open profile page
- switch off the internet
- open main page (tips list page)

I got the `maintenance page` in that way. Not sure why it's not fires all the time. Looks like Brave has other rules for fire an error on `internet disconnected` but the same for the other errors. Maybe that's related to the manually switch off the internet.

**Comment by Liubov-crypto** [comment]
> @Liubov-crypto try to reproduce it that way in Brave:
> 
> * open profile page
> * switch off the internet
> * open main page (tips list page)
> 
> I got the `maintenance page` in that way. Not sure why it's not fires all the time. Looks like Brave has other rules for fire an error on `internet disconnected` but the same for the other errors. Maybe that's related to the manually switch off the internet.

Works in this way. But after connecting to the Internet, the user remains on the maintenance page. No automatic redirection to the main page.

**Comment by shapkarin** [comment]
> > @Liubov-crypto try to reproduce it that way in Brave:
> > 
> > * open profile page
> > * switch off the internet
> > * open main page (tips list page)
> > 
> > I got the `maintenance page` in that way. Not sure why it's not fires all the time. Looks like Brave has other rules for fire an error on `internet disconnected` but the same for the other errors. Maybe that's related to the manually switch off the internet.
> 
> Works in this way. But after connecting to the Internet, the user remains on the maintenance page. No automatic redirection to the main page.

Yes, the logic was changed. @davidyuk suggest to do that. But I can try to forward it to the main on connect.

**Comment by shapkarin** [comment]
@Liubov-crypto I guess now it can be forwarded to main page after the maintenance

**Comment by Liubov-crypto** [comment]
> @Liubov-crypto I guess now it can be forwarded to main page after the maintenance

now redirects to the main page, but periodically redirects to https://feature-refact-backend.aggregator.z52da5wt.xyz/maintenance 

also, I can't stay on my profile page, redirection to the main page after a few seconds

File: src/utils/backend.js (hunk 1 above)
Comment (mradkov) [reply src/utils/backend.js]: And how the user gets redirected back to the feed when the maintenance is finished?

**Comment by shapkarin** [comment]
@Liubov-crypto thanks. It's a bug

File: src/utils/backend.js (hunk 1 above)
Comment (shapkarin) [reply src/utils/backend.js]: > When does the maintenance page return back to the main feed, if you comment this out?

@mradkov right, need to solve that

Comment (shapkarin) [reply src/utils/backend.js]: @mradkov not sure that it's possible now

Comment (shapkarin) [reply src/utils/backend.js]: @mradkov I guess I solved that. Need to recheck.

**Comment by shapkarin** [comment]
@Liubov-crypto after my last commit it should redirects to the feed page (the main page). Please, check.

File: src/utils/backend.js (hunk 1 above)
Comment (shapkarin) [reply src/utils/backend.js]: @mradkov I don't like all this PR at all but I have no idea how to easily achieve the SSR (remove import store explicitly) 
This should work as a temporary solution. It also redirects to the feed page with the last commit https://github.com/aeternity/superhero-ui/pull/1003/commits/6e1db153b3e122c5234b1af198d47040edfa4687

**Comment by Liubov-crypto** [comment]
> @Liubov-crypto after my last commit it should redirects to the feed page (the main page). Please, check.

Followed these steps: 
- open profile page
- switch off the internet
- open main page (tips list page)
- turn on the internet
- after a while the app should show the feed list

Got the issue that the main page redirected to the maintenance page after a while. 
I recorded a video, the same behaviour in FF and Brave: 


https://user-images.githubusercontent.com/69896204/106155007-e0ce6000-6188-11eb-8540-6243c3f0e404.mov

**Comment by shapkarin** [comment]
@Liubov-crypto some loop redirects... after a while it's fine. I will investigate.

**Comment by shapkarin** [comment]
@Liubov-crypto pushed the commit, I checked, it's fine with Brave now. Cam you confirm?

**Comment by Liubov-crypto** [comment]
> @Liubov-crypto pushed the commit, I checked, it's fine with Brave now. Can you confirm?

Followed these steps:

open profile page
switch off the internet
open main page (tips list page)
turn on the internet
after a while the app should show the feed list

Now everything looks good, no redirection to maintenance. 
btw, there is some delay in loading the feed in FF.

**Review by mradkov — DISMISSED** [review]
As a way to get it working, this look acceptable. Though we need to get rid of all EventBus usages in the future - logged an issue for this [here](https://github.com/aeternity/superhero-ui/issues/1006).

**Comment by Liubov-crypto** [comment]
Btw, there is no maintenance banner in Safari, just this: 
саф

After turning on the Internet, you need to refresh the page so that the feed appears

**Pull request closed without merging**